src/Microsoft.Health.Dicom.Api/Web/AspNetCoreMultipartReader.cs

```diff
@@ -125,9 +129,13 @@ public async Task<MultipartBodyPart> ReadNextBodyPartAsync(CancellationToken can
             // The stream must be consumed before the next ReadNextSectionAsync is called.
             // Also, the stream returned by the MultipartReader is not seekable. We need to make
             // it seekable so that we can process the stream multiple times.
-            return new MultipartBodyPart(
+            MultipartBodyPart part = new MultipartBodyPart(
                 contentType,
                 await _seekableStreamConverter.ConvertAsync(section.Body, cancellationToken));
+
+            _dicomRequestContextAccessor.RequestContext.RequestParts++;
```

Comment:
Can we track the billing informations in service layer? StoreService/StoreOrchestrator can give you the same

Reply:
Yeah that's a good idea - Moved to StoreService